packages/ember-views/lib/views/view.js

```diff
@@ -1791,7 +1791,7 @@ var View = CoreView.extend({
     Ember.assert("Only arrays are allowed for 'classNameBindings'", typeOf(this.classNameBindings) === 'array');
     this.classNameBindings = emberA(this.classNameBindings.slice());
 
-    Ember.assert("Only arrays are allowed for 'classNames'", typeOf(this.classNames) === 'array');
+    Ember.assert("Only arrays are allowed for 'classNames'. To bind class names to computed properties, use 'classNameBindings' instead.", typeOf(this.classNames) === 'array');
```

Comment:
The original point of the assertion was that folks would use a space separated string and then not understand why it wouldn't work.

I think the point you are trying to make is good, but I would prefer to not directly reference computed properties.  Something like `Only arrays of static class strings are allowed for 'classNames'. For dynamic classes use `classNameBindings`.` makes roughly the same point, right?